File: solutions/lua/binary/1/binary.lua
local function to_decimal(input)
  d=0
  m=1
  for i=string.len(input),1,-1
  do
    if string.sub(input,i,i) == '1'
    then 
      d=d+m 
    elseif string.sub(input,i,i) ~= '0'
    then
      return 0
    end
    m=m*2
  end
  return d
end

return {
  to_decimal = to_decimal
}

File: solutions/vbnet/binary/1/Binary.vb
Public Class Binary
    Public binary As String
    Public Sub New(newbinary As String)
        binary = newbinary
    End Sub

    Public Function ToDecimal() As Integer
        Dim dec, mul As Integer
        mul = 1
        dec = 0
        For i As Integer = binary.Length() - 1 To 0 Step -1
            If binary.Substring(i, 1) = "1" Then
                dec = dec + mul
            ElseIf binary.Substring(i, 1) <> "0" Then
                Return 0
            End If
            mul = mul * 2
        Next
        Return dec
    End Function
End Class